upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
+        const end = Math.min(start + chunkSize, file.size);
+        const chunk = file.slice(start, end);
+
+        // Creating a new XMLHttpRequest for every chunk (bad practice)
+        var xhr = new XMLHttpRequest();
```

Comment:
Use `const` instead of `var` for the `xhr` variable as it is not being reassigned.

```suggestion
const xhr = new XMLHttpRequest();
```